### ISSUE TITLE
When fetching, app crashes if you fetch twice

### ISSUE THREAD

**Issue by tamuseanmiller** [oceanbit-dev/GitShark#51]
I was able to reproduce this over and over. Here are the steps:

- Open a repo, tap the overflow menu, and then fetch->fetch.
- After you do this if it fails, retry then go back as if to cancel (Which there should probably be a button for).
- Tap the overflow menu again and then fetch->fetch.

This should be where it crashes

**Comment by crutchcorn** [comment]
I've been having the hardest time reproducing this bug, but finally managed to do so.

A few things on this:

1) I am rewriting how `fetch` is handled in GitShark Android, so we'll have this bug moving forward

2) However, all current repos seem to be stuck in a really weird limbo state. Once this bug is "fixed", I'm unable to patch old repos that were setup with the old "fetch" code

3) However, this shouldn't cause the app to crash anymore, even with the broken code. It will simply throw an error screen that we've programmed in now. (it's ugly, but we're working on fixing the UI of this before open-beta)

TL;DR we don't have a fix out right this minute, but we should have a fix out within a day or two. Once we do, you'll need to do a `fetch all` and then another `fetch all` to fix this issue
